packages/devextreme/js/__internal/ui/collection/collection_widget.edit.ts

```diff
@@ -224,6 +225,7 @@ class CollectionWidget<
     const { itemsGetter } = this._editStrategy;
     const { selectionMode, maxFilterLengthInRequest } = this.option();
 
+    // @ts-expect-error TItem
     this._selection = new Selection({
```

Comment:
```suggestion
    this._selection = new Selection<TItem>({
```
this solves ts issues in callbacks and in other properties as well

Reply:
fixed in: https://github.com/DevExpress/DevExtreme/pull/30768/commits/7dfa3fcfb4071bac6bb77c47615c9dd41725df4d